servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublishAndSubscribeOnCompletables.java

```diff
@@ -64,21 +64,22 @@ static Completable subscribeOn(Completable original, Executor executor) {
      */
     private static final class PublishOn extends TaskBasedAsyncCompletableOperator {
 
-        PublishOn(final Completable original, final Executor executor) {
-            super(original, executor);
-        }
-
-        @Override
-        public Subscriber apply(Subscriber subscriber) {
-            return new CompletableSubscriberOffloadedTerminals(subscriber, executor());
+        PublishOn(final Completable original,
+                  final Supplier<? extends BooleanSupplier> shouldOffloadSupplier, final Executor executor) {
+            super(original, shouldOffloadSupplier, executor);
         }
 
         @Override
         void handleSubscribe(final Subscriber subscriber,
                              final AsyncContextMap contextMap, final AsyncContextProvider contextProvider) {
             // re-wrap the subscriber so that async context is restored during offloading.
             Subscriber wrapped = contextProvider.wrapCompletableSubscriber(subscriber, contextMap);
-            super.handleSubscribe(wrapped, contextMap, contextProvider);
+
+            BooleanSupplier shouldOffload = shouldOffloadSupplier();
```

Comment:
looks like https://github.com/apple/servicetalk/pull/1689#discussion_r675885472 was collapsed but still not resolved

> since we are invoking user code here, we should wrap in try/catch like in `SubscribeOn` below.

Reply:
Now checked all 6 spots and all are wrapped as well as all sites where the `BooleanSupplier` is invoked.